.changeset/fair-beds-happen.md

```diff
@@ -0,0 +1,5 @@
+---
+'@urql/core': patch
+---
+
+Make DocumentNode serialization idempotent, closes #1508
```

Comment:
```suggestion
Fix inconsistency in generating keys for `DocumentNode`s, especially when using GraphQL Code Generator, which could cause SSR serialization to fail.
```

Just a minor one here; this goes directly into the changelog, so we make an effort to describe exactly what went on in the change and how it affects users (also bug fixes / critical hot fixes are prefixed by "Fix" so they're highlighted) ✌️